CIScripts/Tests/AgentServiceTests.ps1

```diff
@@ -0,0 +1,258 @@
+        Initialize-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+        Install-Agent -Session $Session
+        Assert-IsAgentServiceRegistered -Session $Session
+
+        New-AgentConfigFile -Session $Session -TestConfiguration $TestConfiguration
+
+        Write-Host "======> When Agent is enabled"
+        Enable-AgentService -Session $Session
+        Assert-IsAgentServiceEnabled -Session $Session
+
+        Write-Host "======> Then Agent state is equal to ENABLE after a while"
+        Start-Sleep -s $WAIT_TIME_FOR_AGENT_SERVICE_IN_SECONDS
+        Assert-IsAgentServiceEnabled -Session $Session
+
+        Write-Host "===> PASSED: Test-AgentServiceEnabling"
+    }
+
+    function Test-AgentServiceDisabling {
+        Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
+               [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
+
+        Write-Host "===> Running: Test-AgentServiceDisabling"
+
+        Write-Host "======> Given Agent is installed and enabled"
+        Clear-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+        Initialize-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+        Install-Agent -Session $Session
+        Assert-IsAgentServiceRegistered -Session $Session
+
+        New-AgentConfigFile -Session $Session -TestConfiguration $TestConfiguration
+
+        Enable-AgentService -Session $Session
+        Assert-IsAgentServiceEnabled -Session $Session
+
+        Write-Host "======> When Agent is disabled"
+        Disable-AgentService -Session $Session
+        Assert-IsAgentServiceDisabled -Session $Session
+
+        Write-Host "======> Then Agent state is equal to DISABLE after a while"
+        Start-Sleep -s $WAIT_TIME_FOR_AGENT_SERVICE_IN_SECONDS
+        Assert-IsAgentServiceDisabled -Session $Session
+
+        Write-Host "===> PASSED: Test-AgentServiceDisabling"
+    }
+
+
+    $AgentServiceTestsTimeTracker.StepQuiet("Test-AgentServiceIsRegisteredAfterInstall", {
+        Test-AgentServiceIsRegisteredAfterInstall -Session $Session -TestConfiguration $TestConfiguration
+    })
+    $AgentServiceTestsTimeTracker.StepQuiet("Test-AgentServiceIsDisabledAfterInstall", {
+        Test-AgentServiceIsDisabledAfterInstall -Session $Session -TestConfiguration $TestConfiguration
+    })
+    $AgentServiceTestsTimeTracker.StepQuiet("Test-AgentServiceIsUnregisteredAfterUninstall", {
+        Test-AgentServiceIsUnregisteredAfterUninstall -Session $Session -TestConfiguration $TestConfiguration
+    })
+    $AgentServiceTestsTimeTracker.StepQuiet("Test-AgentServiceEnabling", {
+        Test-AgentServiceEnabling -Session $Session -TestConfiguration $TestConfiguration
+    })
+    $AgentServiceTestsTimeTracker.StepQuiet("Test-AgentServiceDisabling", {
+        Test-AgentServiceDisabling -Session $Session -TestConfiguration $TestConfiguration
+    })
+
+    # Test cleanup
```

Comment:
If you do not wish to assume any specific order of invoking tests (as you mentioned earlier), please make sure that Agent is installed here (some tests leave it uninstalled).